torch/distributed/algorithms/ddp_comm_hooks/ddp_zero_hook.py

```diff
@@ -46,65 +47,68 @@ def hook_then_zero_fn(
                 gradient bucket.
         """
         fut = hook(state, bucket)
+        if not zero._use_extra_stream:
```

Comment:
I am not sure why I added a `not` here. I think this is wrong and am investigating this currently.